app/assets/javascripts/components/settings/views/add_admin_button.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import Popover from '../../common/popover.jsx';
+import PopoverExpandable from '../../high_order/popover_expandable.jsx';
+import AddAdminForm from '../containers/add_admin_form_container';
+
+class AddAdminButton extends React.Component {
+  constructor() {
```

Comment:
I've never used this pattern before. Why are these steps in the constructor necessary?

Could this be a stateless functional component instead?

Reply:
stateless components can't use refs and `PopoverExpandable` uses refs with this component. In the constructor I am binding `this` to these methods in order for them to have access. That will be a moot point once I convert everything to `React.createClass` though :)